addons/account/static/src/js/bills_tree_upload.js

```diff
@@ -75,6 +75,23 @@ odoo.define('account.bills.tree', function (require) {
             'click .o_button_upload_bill': '_onUpload',
             'change .o_vendor_bill_upload .o_form_binary_form': '_onAddAttachment',
         }),
+        /**
+         * We override _onUpload (from the upload bill mixin) to pass default_journal_id
+         * and default_type in context.
+         *
+         * @override
+         */
+        _onUpload: function (event) {
+            var context = this.initialState.context;
+            if (context.active_model == 'account.move' && context.active_id !== undefined) {
```

Comment:
When will this first condition be true ? (not sure)

Reply:
looks like it was a typo, it should have been `account.journal`(since active_id was having journal's id), but since we carry forward context from `create_invoice_from_attachment` we do not need to override `_onUpload` for `BillsListController`.